### PR TITLE
Danny/verification fix

### PR DESCRIPTION
**Pull request by DannyLeWasTaken**
- Hopefully fix crashing by adding a retry cooldown (max of 5 retries) for writing to the file
- Reject people if they are not in GryphLife organization

### CODE REVIEW

File: src/vertification.ts [hunk 1]
```diff
@@ -265,6 +298,12 @@ export async function handleVerificationDM(client: Client, message: Message) {
         processing_members_code.delete(message.author.id);
         return;
     }
+    if (message.content === "cancel") {
+        // Verification cancelled
+        processing_members_code.delete(message.author.id);
+        await message.reply("Verification code cancelled. Please resend if your email address to get a new one");
```
Comment (Nomalah) [inline src/vertification.ts:304]:
```suggestion
        await message.reply("Verification code cancelled. Please resend your email address to get a new one");
```

File: src/vertification.ts [hunk 2]
```diff
@@ -273,12 +312,13 @@ export async function handleVerificationDM(client: Client, message: Message) {
         // Update spreadsheet
         const user_row = verification_spreadsheet.find(data => data.email === verification_code.email)!;
         user_row.discord_identifier = message.author.tag;
-        pushSpreadsheet();
+        const push_promise = pushSpreadsheet();
```
Comment (Nomalah) [inline src/vertification.ts:315]:
```suggestion
        await pushSpreadsheet();
```

File: src/vertification.ts [hunk 3]
```diff
@@ -273,12 +312,13 @@ export async function handleVerificationDM(client: Client, message: Message) {
...
         processing_members_code.delete(message.author.id);
         await message.reply(`Verification successful! Welcome aboard, ${user_row.name}.`);
 
         const channel = guild.channels.resolve(process.env.VERIFICATION_CHANNEL!) as TextChannel;
         await channel.send(`${message.author.tag} has been successfully verified`);
+        await push_promise;
```
Comment (Nomalah) [inline src/vertification.ts:321]:
```suggestion
```

File: src/vertification.ts [hunk 4]
```diff
@@ -113,8 +120,19 @@ function pushSpreadsheet() {
...
+    let retries: number = 0; // MAX RETRIES - 5
+
     // Write the workbook to the filesystem
-    writeFile(workbook, FILE_PATH);
+    while (retries < 5) {
+        try {
+            writeFile(workbook, FILE_PATH);
+            break;
+        } catch (_) {
+            // Failed to write to file usually due to something writing it already. Override it :D
```
Comment (Nomalah) [inline src/vertification.ts:131]: Maybe log this?

**Review by Nomalah — CHANGES_REQUESTED** [review]

File: src/vertification.ts (hunk 4 above)
Comment (DannyLeWasTaken) [reply src/vertification.ts:131]: Yeah will do

File: src/vertification.ts [hunk 5]
```diff
@@ -245,9 +277,11 @@ export async function handleVerification(message: Message) {
                 message.reply({ content: "Failed to send email address. Make sure your email address is valid." });
                 return;
             });
-        await message.reply({ content: "Please **DM the bot** with a 7 digit code sent to the email address." });
+        await message.reply({ content: "Please **DM the bot** with a 7 digit code sent to the email address. Type `cancel` if you wish to cancel the verification code." });
     } else if (!user_row) {
         await message.reply({ content: `Your email is not registered. You have not submitted your application to the [form](<${FORM_LINK}>).` });
```
Comment (Nomalah) [inline src/vertification.ts:282]: Rephrase it to say, you _may_ have not submitted, or your application _may_ have not been processed yet.

File: src/vertification.ts [hunk 6]
```diff
@@ -245,9 +277,11 @@ export async function handleVerification(message: Message) {
...
+    } else if (user_row.in_gryphlife !== "yes") {
+        await message.reply({ content: `You are not in the [GryphLife](<${GRYPHLIFE_LINK}>) organization.` });
```
Comment (Nomalah) [inline src/vertification.ts:284]: Rephrase to say you have not joined, or your request and not been accepted yet.

**Review by Nomalah — APPROVED** [review]
Could you adjust the wording on those responses before merging?
